src/constraint/CollisionFree.cpp

```diff
@@ -27,31 +27,63 @@ statespace::StateSpacePtr CollisionFree::getStateSpace() const
 
 //==============================================================================
 bool CollisionFree::isSatisfied(
-    const aikido::statespace::StateSpace::State* _state) const
+    const aikido::statespace::StateSpace::State* _state,
+    TestableOutcome* outcome) const
 {
   auto skelStatePtr = static_cast<const aikido::statespace::dart::
                                       MetaSkeletonStateSpace::State*>(_state);
   mStatespace->setState(skelStatePtr);
 
   bool collision = false;
   dart::collision::CollisionResult collisionResult;
-  for (auto groups : mGroupsToPairwiseCheck)
+  for (const auto& groups : mGroupsToPairwiseCheck)
   {
     collision = mCollisionDetector->collide(
         groups.first.get(),
         groups.second.get(),
         mCollisionOptions,
         &collisionResult);
+
     if (collision)
+    {
+      if (outcome != nullptr)
+      {
+        const auto& collisionOutcome
+            = dynamic_cast<CollisionFreeOutcome*>(outcome);
+        if (collisionOutcome == nullptr)
+          throw std::invalid_argument(
+              "TestableOutcome pointer is not of type CollisionFreeOutcome.");
+        const auto& contacts = collisionResult.getContacts();
+        for (const auto& elem : contacts)
+        {
+          collisionOutcome->markPairwiseContact(elem);
+        }
+      }
       return false;
+    }
   }
 
-  for (auto group : mGroupsToSelfCheck)
+  for (const auto& group : mGroupsToSelfCheck)
   {
     collision = mCollisionDetector->collide(
         group.get(), mCollisionOptions, &collisionResult);
     if (collision)
+    {
+      if (outcome != nullptr)
+      {
+        const auto& collisionOutcome
```

Comment:
I think this should just be `auto collisionOutcome`, for the same reason as above.